src/app/lecturer-page/student-review-form/student-review-form.component.ts

```diff
@@ -0,0 +1,191 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+import { Store } from '@ngrx/store';
+import { concat, Observable, of } from 'rxjs';
+import { catchError, map } from 'rxjs/operators';
+import { ReviewService } from 'src/app/service/review.service';
+import { Stream } from 'src/app/shared/stream';
+import { Student } from 'src/app/shared/student';
+import { selectStreams } from 'src/app/store/stream/stream.selectors';
+import { selectStudents } from 'src/app/store/students/students.selectors';
+
+@Component({
+  selector: 'app-student-review-form',
+  templateUrl: './student-review-form.component.html',
+  styleUrls: ['./student-review-form.component.scss']
+})
+export class StudentReviewFormComponent implements OnInit {
+  reviewForm!: FormGroup;
+
+  students$!: Observable<Student[]>;
+  streams$!: Observable<Stream[]>;
+
+  studentId$!: Observable<Student>;
+  streamId$!: Observable<number>;
+
+  public communicationCharsRemaining$: Observable<number> | undefined;
+  public abilityToLearnCharsRemaining$: Observable<number> | undefined;
+  public extraMileCharsRemaining$: Observable<number> | undefined;
+  public motivationCharsRemaining$: Observable<number> | undefined;
+  public directionCharsRemaining$: Observable<number> | undefined;
+  public overallCharsRemaining$: Observable<number> | undefined;
+
+  defaultSliderValue = 5;
+  maxChars = 255;
```

Comment:
```suggestion
  public readonly defaultSliderValue = 5;
  public readonly maxChars = 255;
```
and move these to the very top of the component as these are constants :)